RecoParticleFlow/PFClusterTools/interface/LinkByRecHit.h

```diff
@@ -12,12 +14,100 @@ class LinkByRecHit {
+
   //tests association between a track and a cluster by rechit
   static double testTrackAndClusterByRecHit(const reco::PFRecTrack& track,
                                             const reco::PFCluster& cluster,
                                             bool isBrem = false,
                                             bool debug = false);
 
+  static double testTrackAndClusterByRecHit(size_t icluster,
+                                            std::set<size_t> cluster_rechits,
+                                            edm::soa::TableView<edm::soa::col::pf::cluster::Eta,
+                                                                edm::soa::col::pf::cluster::Phi,
+                                                                edm::soa::col::pf::cluster::Posz,
+                                                                edm::soa::col::pf::cluster::Layer,
+                                                                edm::soa::col::pf::cluster::FracsNbr> cluster_table,
+
+                                            edm::soa::TableView<edm::soa::col::pf::rechit::DetIdValue,
+                                                                edm::soa::col::pf::rechit::Fraction,
+                                                                edm::soa::col::pf::rechit::Eta,
+                                                                edm::soa::col::pf::rechit::Phi,
+                                                                edm::soa::col::pf::rechit::Posx,
+                                                                edm::soa::col::pf::rechit::Posy,
+                                                                edm::soa::col::pf::rechit::Posz,
+                                                                edm::soa::col::pf::rechit::Corner0x,
+                                                                edm::soa::col::pf::rechit::Corner0y,
+                                                                edm::soa::col::pf::rechit::Corner0z,
+                                                                edm::soa::col::pf::rechit::Corner1x,
+                                                                edm::soa::col::pf::rechit::Corner1y,
+                                                                edm::soa::col::pf::rechit::Corner1z,
+                                                                edm::soa::col::pf::rechit::Corner2x,
+                                                                edm::soa::col::pf::rechit::Corner2y,
+                                                                edm::soa::col::pf::rechit::Corner2z,
+                                                                edm::soa::col::pf::rechit::Corner3x,
+                                                                edm::soa::col::pf::rechit::Corner3y,
+                                                                edm::soa::col::pf::rechit::Corner3z,
```

Comment:
seeing how these have to be scattered into N variables to be then gathered back into array of size N, is suggestive that these should be arrays explicitly

Reply:
let me try putting std::array in SOA...

Reply:
If this sort of multiple lines can be simplified to fit into a fewer lines using arrays, I think it will be great.